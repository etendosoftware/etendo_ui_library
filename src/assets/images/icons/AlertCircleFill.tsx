import React from 'react';
import { ImageStyle, ColorValue } from 'react-native';
import { Svg, Path } from 'react-native-svg';
import { DEFAULT_COLOR_THEME } from '../../../styles/colors';
import { sizeSvg } from '../../../helpers/svg_utils';

export interface SvgImageProps {
  style?: ImageStyle;
  fill?: ColorValue;
}

export const AlertCircleFill = ({
  style,
  fill = DEFAULT_COLOR_THEME,
}: SvgImageProps) => {
  const width = sizeSvg(style?.width, 24);
  const height = sizeSvg(style?.height, 24);

  return (
    <Svg width={width} height={height} viewBox="0 0 24 24" fill="none">
      <Path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM13 8C13 7.44772 12.5523 7 12 7C11.4477 7 11 7.44772 11 8V12C11 12.5523 11.4477 13 12 13C12.5523 13 13 12.5523 13 12V8ZM12 15C11.4477 15 11 15.4477 11 16C11 16.5523 11.4477 17 12 17H12.01C12.5623 17 13.01 16.5523 13.01 16C13.01 15.4477 12.5623 15 12.01 15H12Z"
        fill={fill}
      />
    </Svg>
  );
};
