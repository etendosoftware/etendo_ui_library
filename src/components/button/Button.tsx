import React, { useState } from 'react';
import {
  ActivityIndicator,
  Pressable,
  Text,
  TouchableOpacity,
} from 'react-native';
import {
  NEUTRAL_50,
  PRIMARY_100,
  SECONDARY_300,
  TERTIARY_50,
  TERTIARY_800,
} from '../../styles/colors';
import { ButtonProps, ButtonStyleType, IconStyleProps } from './Button.types';
import { ButtonStyleVariant, styles } from './Button.styles';

const Button = ({
  text,
  onPress,
  fontSize = 16,
  typeStyle,
  disabled,
  width = 'auto',
  height,
  iconLeft,
  iconRight,
  paddingHorizontal = 12,
  paddingVertical = 12,
  loading = false,
}: ButtonProps) => {
  const [isHovered, setIsHovered] = useState(false);

  const getHoveredBackgroundColor = (backgroundColor: ButtonStyleType) => {
    switch (backgroundColor) {
      case 'primary':
        return TERTIARY_800;
      case 'secondary':
        return SECONDARY_300;
      case 'terciary':
        return TERTIARY_50;
      case 'whiteBorder':
        return PRIMARY_100;
    }
  };
  const getHoveredTextColorAndViewColor = (typeStyle: ButtonStyleType) => {
    switch (typeStyle) {
      case 'white':
        return TERTIARY_800;
      case 'whiteBorder':
        return NEUTRAL_50;
      case 'primary':
        return NEUTRAL_50;
      default:
        return PRIMARY_100;
    }
  };

  const stateStyleContainer = (pressed: boolean) => {
    if (disabled) {
      return ButtonStyleVariant[typeStyle].containerDisabled;
    }
    if (pressed && !loading) {
      const hoveredBackgroundColor = getHoveredBackgroundColor(typeStyle);
      return {
        ...ButtonStyleVariant[typeStyle].container,
        opacity: 0.7,
        backgroundColor: hoveredBackgroundColor,
      };
    }
    if (isHovered && !loading) {
      const hoveredBackgroundColor = getHoveredBackgroundColor(typeStyle);
      return {
        ...ButtonStyleVariant[typeStyle].container,
        backgroundColor: hoveredBackgroundColor,
      };
    }
    return ButtonStyleVariant[typeStyle].container;
  };
  const handleHoverIn = () => {
    setIsHovered(true);
  };

  const handleHoverOut = () => {
    setIsHovered(false);
  };

  const stateStyleText = () => {
    if (loading && !iconLeft && !iconRight) {
      return { color: 'transparent' };
    }
    if (disabled) {
      return ButtonStyleVariant[typeStyle].textDisabled;
    }
    if (isHovered && !loading) {
      const hoveredTextColor = getHoveredTextColorAndViewColor(typeStyle);
      if (hoveredTextColor) {
        return {
          ...ButtonStyleVariant[typeStyle].text,
          color: hoveredTextColor,
        };
      }
    }
    return ButtonStyleVariant[typeStyle].text;
  };

  const stateStyleIcon = () => {
    if (disabled) {
      return ButtonStyleVariant[typeStyle].imageDisabled;
    }
    if (isHovered && !loading) {
      return isHovered
        ? getHoveredTextColorAndViewColor(typeStyle)
        : ButtonStyleVariant[typeStyle].imageColor;
    }
    return ButtonStyleVariant[typeStyle].imageColor;
  };

  const stateSizeIndicator = (iconComponent: React.ReactElement) => {
    if (iconComponent?.props?.style?.height) {
      return iconComponent?.props?.style?.height;
    }
    if (iconComponent?.props?.style?.width) {
      return iconComponent?.props?.style?.width;
    }
    return 16;
  };

  const renderIcon = (
    icon?: React.ReactElement,
    { marginRight = 0, marginLeft = 0 }: IconStyleProps = {},
  ) => {
    if (!icon) return null;

    const iconSize = stateSizeIndicator(icon);
    const iconColor = stateStyleIcon();

    if (loading) {
      return (
        <ActivityIndicator
          size={iconSize}
          color={iconColor}
          style={{ marginRight, marginLeft }}
        />
      );
    }

    return React.cloneElement(icon, {
      style: {
        width: icon?.props?.style?.width || 16,
        height: icon?.props?.style?.height || 16,
        marginRight,
        marginLeft,
      },
      fill: iconColor,
    });
  };

  const handleOnPress = () => {
    if (onPress) {
      onPress();
    }
  };
  return (
    <TouchableOpacity>
      <Pressable
        onHoverIn={handleHoverIn}
        onHoverOut={handleHoverOut}
        style={({ pressed }) => [
          stateStyleContainer(pressed),
          { height, width, paddingHorizontal, paddingVertical },
        ]}
        onPress={handleOnPress}>
        {renderIcon(iconLeft, {
          marginRight: iconLeft && text ? 8 : 0,
        })}
        {loading && !iconLeft && !iconRight && (
          <ActivityIndicator
            size={fontSize}
            color={stateStyleIcon()}
            style={styles.indicatorCenterStyle}
          />
        )}
        <Text style={[stateStyleText(), { fontSize }]}>{text}</Text>
        {renderIcon(iconRight, {
          marginLeft: iconRight && text ? 8 : 0,
        })}
      </Pressable>
    </TouchableOpacity>
  );
};

export default Button;
