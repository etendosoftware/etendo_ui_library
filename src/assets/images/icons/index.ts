import { ActivityIcon } from './ActivityIcon';
import { AirplayIcon } from './AirplayIcon';
import { AlertCicleIcon } from './AlertCicleIcon';
import { AlertCircleFill } from './AlertCircleFill';
import { AlertOctagonIcon } from './AlertOctagonIcon';
import { AlertTriangleIcon } from './AlertTriangleIcon';
import { AlignCenterIcon } from './AlignCenterIcon';
import { AlignJustifyIcon } from './AlignJustifyIcon';
import { AlignLeftIcon } from './AlignLeftIcon';
import { AlignRightIcon } from './AlignRightIcon';
import { AnchorIcon } from './AnchorIcon';
import { ApertureIcon } from './ApertureIcon';
import { ArchiveIcon } from './ArchiveIcon';
import { ArrowDownCircleIcon } from './ArrowDownCircleIcon';
import { ArrowDownIcon } from './ArrowDownIcon';
import { ArrowDownLeftIcon } from './ArrowDownLeftIcon';
import { ArrowDownRightIcon } from './ArrowDownRightIcon';
import { ArrowLeftCircleIcon } from './ArrowLeftCircleIcon';
import { ArrowLeftIcon } from './ArrowLeftIcon';
import { ArrowRightCircleIcon } from './ArrowRightCircleIcon';
import { ArrowRightIcon } from './ArrowRightIcon';
import { ArrowUpCircleIcon } from './ArrowUpCircleIcon';
import { ArrowUpIcon } from './ArrowUpIcon';
import { ArrowUpLeftIcon } from './ArrowUpLeftIcon';
import { ArrowUpRightIcon } from './ArrowUpRightIcon';
import { AtSignIcon } from './AtSignIcon';
import { AwardIcon } from './AwardIcon';
import { BarChartIcon } from './BarChartIcon';
import { BarChartIcon2 } from './BarChartIcon2';
import { BateryIcon } from './BateryIcon';
import { BatteryChargingIcon } from './BatteryChargingIcon';
import { BellIcon } from './BellIcon';
import { BellOffIcon } from './BellOffIcon';
import { BluetoothIcon } from './BluetoothIcon';
import { BoldIcon } from './BoldIcon';
import { BookIcon } from './BookIcon';
import { BookOpenIcon } from './BookOpenIcon';
import { BookmarkIcon } from './BookmarkIcon';
import { BoxIcon } from './BoxIcon';
import { BriefCaseIcon } from './BriefCaseIcon';
import { CalendarIcon } from './CalendarIcon';
import { CameraIcon } from './CameraIcon';
import { CameraOffIcon } from './CameraOffIcon';
import { CastIcon } from './CastIcon';
import { CheckCircleFillIcon } from './CheckCircleFillIcon';
import { CheckCircleIcon } from './CheckCircleIcon';
import { CheckIcon } from './CheckIcon';
import { CheckSquareicon } from './CheckSquareicon';
import { ChevronDownIcon } from './ChevronDownIcon';
import { ChevronLeftIcon } from './ChevronLeftIcon';
import { ChevronRightIcon } from './ChevronRightIcon';
import { ChevronUpIcon } from './ChevronUpIcon';
import { ChevronsDownIcon } from './ChevronsDownIcon';
import { ChevronsRightIcon } from './ChevronsRightIcon';
import { ChevronsLeftIcon } from './ChevronsLeftIcon';
import { ChevronsUpIcon } from './ChevronsUpIcon';
import { ChromeIcon } from './ChromeIcon';
import { CircleIcon } from './CircleIcon';
import { ClearIcon } from './ClearIcon';
import { ClipboardIcon } from './ClipboardIcon';
import { ClockIcon } from './ClockIcon';
import { CloudDrizzleIcon } from './CloudDrizzleIcon';
import { CloudRainIcon } from './CloudRainIcon';
import { CloudIcon } from './CloudIcon';
import { CloudLightingIcon } from './CloudLightingIcon';
import { CloudOffIcon } from './CloudOffIcon';
import { CloudSnowIcon } from './CloudSnowIcon';
import { CodeIcon } from './CodeIcon';
import { CodepenIcon } from './CodepenIcon';
import { CodesandboxIcon } from './CodesandboxIcon';
import { CoffeIcon } from './CoffeIcon';
import { ColumnsIcon } from './ColumnsIcon';
import { CommandIcon } from './CommandIcon';
import { CompassIcon } from './CompassIcon';
import { CopyIcon } from './CopyIcon';
import { CornerDownLeftIcon } from './CornerDownLeftIcon';
import { CornerDownRightIcon } from './CornerDownRightIcon';
import { CornerLeftDownIcon } from './CornerLeftDownIcon';
import { CornerLeftUpIcon } from './CornerLeftUpIcon';
import { CornerRightDownIcon } from './CornerRightDownIcon';
import { CornerRightUpIcon } from './CornerRightUpIcon';
import { CornerUpLeftIcon } from './CornerUpLeftIcon';
import { CornerUpRight } from './CornerUpRight';
import { CpuIcon } from './CpuIcon';
import { CreditCardIcon } from './CreditCardIcon';
import { CropIcon } from './CropIcon';
import { CrosshairIcon } from './CrosshairIcon';
import { DatabaseIcon } from './DatabaseIcon';
import { DeleteIcon } from './DeleteIcon';
import { DiscIcon } from './DiscIcon';
import { DivideCircleIcon } from './DivideCircleIcon';
import { DivideIcon } from './DivideIcon';
import { DivideSquareIcon } from './DivideSquareIcon';
import { DollarIcon } from './DollarIcon';
import { DownloadCloudIcon } from './DownloadCloudIcon';
import { DownloadIcon } from './DownloadIcon';
import { DribbbleIcon } from './DribbbleIcon';
import { DropletIcon } from './DropletIcon';
import { Edit2Icon } from './Edit2Icon';
import { Edit3Icon } from './Edit3Icon';
import { EditIcon } from './EditIcon';
import { ExternalLinkIcon } from './ExternalLinkIcon';
import { EyeIcon } from './EyeIcon';
import { EyeOffIcon } from './EyeOffIcon';
import { FacebookIcon } from './FacebookIcon';
import { FastForwardIcon } from './FastForwardIcon';
import { FeatherIcon } from './FeatherIcon';
import { FigmaIcon } from './FigmaIcon';
import { FileIcon } from './FileIcon';
import { FileTextIcon } from './FileTextIcon';
import { FileMinusIcon } from './FileMinusIcon';
import { FilePlusIcon } from './FilePlusIcon';
import { FilmIcon } from './FilmIcon';
import { FilterIcon } from './FilterIcon';
import { FlagIcon } from './FlagIcon';
import { FolderIcon } from './FolderIcon';
import { FolderMinusIcon } from './FolderMinusIcon';
import { FolderPlusIcon } from './FolderPlusIcon';
import { FramerIcon } from './FramerIcon';
import { FrownIcon } from './FrownIcon';
import { GiftIcon } from './GiftIcon';
import { GitBranchIcon } from './GitBranchIcon';
import { GitCommitIcon } from './GitCommitIcon';
import { GitMergeIcon } from './GitMergeIcon';
import { GitPullRequestIcon } from './GitPullRequestIcon';
import { GithubIcon } from './GithubIcon';
import { GitlabIcon } from './GitlabIcon';
import { GlobeIcon } from './GlobeIcon';
import { GridIcon } from './GridIcon';
import { HardDriveIcon } from './HardDriveIcon';
import { HashIcon } from './HashIcon';
import { HeadponesIcon } from './HeadponesIcon';
import { HeartIcon } from './HeartIcon';
import { HelpCircleIcon } from './HelpCircleIcon';
import { HexagonIcon } from './HexagonIcon';
import { HomeIcon } from './HomeIcon';
import { IconInfo } from './IconInfo';
import { ImageIcon } from './ImageIcon';
import { InboxIcon } from './InboxIcon';
import { InstagramIcon } from './InstagramIcon';
import { ItalicICon } from './ItalicICon';
import { KeyIcon } from './KeyIcon';
import { LayerIcon } from './LayerIcon';
import { LayoutIcon } from './LayoutIcon';
import { LifeBouyIcon } from './LifeBouyIcon';
import { Link2Icon } from './Link2Icon';
import { LinkIcon } from './LinkIcon';
import { LinkedinIcon } from './LinkedinICon';
import { ListIcon } from './ListIcon';
import { LoaderIcon } from './LoaderIcon';
import { LockIcon } from './LockIcon';
import { LogInIcon } from './LogInIcon';
import { LogOutIcon } from './LogOutIcon';
import { MailIcon } from './MailIcon';
import { MapIcon } from './MapIcon';
import { MapPinIcon } from './MapPinIcon';
import { Maximize2Icon } from './Maximize2Icon';
import { MaximizeIcon } from './MaximizeIcon';
import { MehIcon } from './MehIcon';
import { MenuIcon } from './MenuIcon';
import { MessageCircleIcon } from './MessageCircleIcon';
import { MessageSquareIcon } from './MessageSquareIcon';
import { MicIcon } from './MicIcon';
import { MicOffIcon } from './MicOffIcon';
import { Minimize2Icon } from './Minimize2Icon';
import { MinimizeIcon } from './MinimizeIcon';
import { MinusCircleIcon } from './MinusCircleIcon';
import { MinusIcon } from './MinusIcon';
import { MinusSquareIcon } from './MinusSquareIcon';
import { MonitorIcon } from './MonitorIcon';
import { MoonIcon } from './MoonIcon';
import { MoreHorizontalIcon } from './MoreHorizontalIcon';
import { MoreVerticalIcon } from './MoreVerticalIcon';
import { MousePointerIcon } from './MousePointerIcon';
import { MoveIcon } from './MoveIcon';
import { MusicIcon } from './MusicIcon';
import { Navigation2Icon } from './Navigation2Icon';
import { NavigationIcon } from './NavigationIcon';
import { OctagonIcon } from './OctagonIcon';
import { PackageIcon } from './PackageIcon';
import { PaperclipIcon } from './PaperclipIcon';
import { PauseClrcleIcon } from './PauseClrcleIcon';
import { PauseIcon } from './PauseIcon';
import { PenTool } from './PenTool';
import { PercentIcon } from './PercentIcon';
import { PhoneCallIcon } from './PhoneCallIcon';
import { PhoneForwardedIcon } from './PhoneForwardedIcon';
import { PhoneIcon } from './PhoneIcon';
import { PhoneIncomingIcon } from './PhoneIncomingIcon';
import { PhoneMissedIcon } from './PhoneMissedIcon';
import { PhoneOffIcon } from './PhoneOffIcon';
import { PhoneOutgoingIcon } from './PhoneOutgoingIcon';
import { PieChartIcon } from './PieChartIcon';
import { PlayCircleIcon } from './PlayCircleIcon';
import { PlayIcon } from './PlayIcon';
import { PlusCircleIcon } from './PlusCircleIcon';
import { PlusIcon } from './PlusIcon';
import { PlusSquareIcon } from './PlusSquareIcon';
import { PowerIcon } from './PocketIcon';
import { PocketIcon } from './PowerIcon';
import { PrinterIcon } from './PrinterIcon';
import { RadioIcon } from './RadioIcon';
import { RefreshCcvIcon } from './RefreshCcvIcon';
import { RefreshCvIcon } from './RefreshCvIcon';
import { RepeatIcon } from './RepeatIcon';
import { RewindIcon } from './RewindIcon';
import { RotateCcwIcon } from './RotateCcwIcon';
import { RotateCwIcon } from './RotateCwIcon';
import { RssIcon } from './RssIcon';
import { SaveIcon } from './SaveIcon';
import { ScissorsIcon } from './ScissorsIcon';
import { SearchIcon } from './SearchIcon';
import { SendIcon } from './SendIcon';
import { ServerIcon } from './ServerIcon';
import { SettingIcon } from './SettingIcon';
import { Share2Icon } from './Share2Icon';
import { ShareIcon } from './ShareIcon';
import { ShieldIcon } from './ShieldIcon';
import { ShieldOffIcon } from './ShieldOffIcon';
import { ShoppingBagIcon } from './ShoppingBagIcon';
import { ShoppingCartIcon } from './ShoppingCartIcon';
import { ShuffleIcon } from './ShuffleIcon';
import { SidebarIcon } from './SidebarIcon';
import { SkipbackIcon } from './SkipbackIcon';
import { SkipforewardIcon } from './SkipforewardIcon';
import { SlackIcon } from './SlackIcon';
import { SlashIcon } from './SlashIcon';
import { SlidersIcon } from './SlidersIcon';
import { SmartphoneIcon } from './SmartphoneIcon';
import { SmileIcon } from './SmileIcon';
import { SpeakerIcon } from './SpeakerIcon';
import { SquareIcon } from './SquareIcon';
import { StarIcon } from './StarIcon';
import { SparksIcon } from './SparksIcon';
import { StopCircleIcon } from './StopCircleIcon';
import { SunIcon } from './SunIcon';
import { SunriseIcon } from './SunriseIcon';
import { SunsetIcon } from './SunsetIcon';
import { TableIcon } from './TableIcon';
import { TabletIcon } from './TabletIcon';
import { TagIcon } from './TagIcon';
import { TargetIcon } from './TargetIcon';
import { TerminalIcon } from './TerminalIcon';
import { ThermometerIcon } from './ThermometerIcon';
import { ThumbsDownIcon } from './ThumbsDownIcon';
import { ThumbsUpIcon } from './ThumbsUpIcon';
import { ToggleLeftIcon } from './ToggleLeftIcon';
import { ToggleRightIcon } from './ToggleRightIcon';
import { ToolIcon } from './ToolIcon';
import { Trash2Icon } from './Trash2Icon';
import { TrashIcon } from './TrashIcon';
import { TrelloIcon } from './TrelloIcon';
import { TrendingDownIcon } from './TrendingDownIcon';
import { TrendingUpIcon } from './TrendingUpIcon';
import { TriangleIcon } from './TriangleIcon';
import { TruckIcon } from './TruckIcon';
import { TvIcon } from './TvIcon';
import { TwitchIcon } from './TwitchIcon';
import { TwitterIcon } from './TwitterIcon';
import { TypeIcon } from './TypeIcon';
import { UmrbrellaIcon } from './UmrbrellaIcon';
import { UnderlineIcon } from './UnderlineIcon';
import { UnlockIcon } from './UnlockIcon';
import { UploadCloudIcon } from './UploadCloudIcon';
import { UploadIcon } from './UploadIcon';
import { UserCheckIcon } from './UserCheckIcon';
import { UserMinusIcon } from './UserMinusIcon';
import { UserPlusIcon } from './UserPlusIcon';
import { UserXIcon } from './UserXIcon';
import { UserIcon } from './UserIcon';
import { UsersIcon } from './UsersIcon';
import { VideoIcon } from './VideoIcon';
import { VideoOffIcon } from './VideoOffIcon';
import { VoiceMailIcon } from './VoiceMailIcon';
import { Volumen1Icon } from './Volumen1Icon';
import { Volumen2Icon } from './Volumen2Icon';
import { VolumenIcon } from './VolumenIcon';
import { VolumenXIcon } from './VolumenXIcon';
import { WatchIcon } from './WatchIcon';
import { WiffiOffIcon } from './WiffiOffIcon';
import { WifiIcon } from './WifiIcon';
import { WindIcon } from './WindIcon';
import { XCircleFillIcon } from './XCircleFillIcon';
import { XCircleIcon } from './XCircleIcon';
import { XIcon } from './XIcon';
import { XOctoganIcon } from './XOctoganIcon';
import { XSquareIcon } from './XSquareIcon';
import { YoutubeIcon } from './YoutubeIcon';
import { ZapIcon } from './ZapIcon';
import { ZapOffIcon } from './ZapOffIcon';
import { ZapOutIcon } from './ZapOutIcon';
import { ZoomInIcon } from './ZoomInIcon';

export {
  ActivityIcon,
  ArrowUpRightIcon,
  ChevronDownIcon,
  CornerDownLeftIcon,
  ExternalLinkIcon,
  GitMergeIcon,
  LinkIcon,
  MoreVerticalIcon,
  PlusIcon,
  ShuffleIcon,
  ToggleLeftIcon,
  AirplayIcon,
  AtSignIcon,
  ChevronLeftIcon,
  CornerLeftDownIcon,
  EyeIcon,
  GitPullRequestIcon,
  ListIcon,
  MousePointerIcon,
  PlusSquareIcon,
  SidebarIcon,
  ToggleRightIcon,
  UserXIcon,
  AlertCicleIcon,
  AwardIcon,
  ChevronRightIcon,
  CornerLeftUpIcon,
  EyeOffIcon,
  GlobeIcon,
  LoaderIcon,
  MoveIcon,
  PocketIcon,
  SkipbackIcon,
  VideoIcon,
  AlertCircleFill,
  BarChartIcon2,
  ChevronsDownIcon,
  CornerRightDownIcon,
  FacebookIcon,
  ClearIcon,
  GridIcon,
  LockIcon,
  MusicIcon,
  PowerIcon,
  SkipforewardIcon,
  ToolIcon,
  VideoOffIcon,
  AlertOctagonIcon,
  BarChartIcon,
  ChevronsRightIcon,
  CornerRightUpIcon,
  FastForwardIcon,
  HardDriveIcon,
  LogInIcon,
  Navigation2Icon,
  PrinterIcon,
  SlackIcon,
  Trash2Icon,
  VoiceMailIcon,
  AlertTriangleIcon,
  BateryIcon,
  ChevronsUpIcon,
  CornerUpLeftIcon,
  FeatherIcon,
  HashIcon,
  LogOutIcon,
  NavigationIcon,
  RadioIcon,
  SlashIcon,
  TrashIcon,
  Volumen1Icon,
  AlignCenterIcon,
  BatteryChargingIcon,
  ChevronUpIcon,
  CornerUpRight,
  FigmaIcon,
  HeadponesIcon,
  MailIcon,
  OctagonIcon,
  RefreshCcvIcon,
  SlidersIcon,
  TrelloIcon,
  Volumen2Icon,
  AlignJustifyIcon,
  BellIcon,
  ChromeIcon,
  CpuIcon,
  FileIcon,
  HeartIcon,
  MapIcon,
  PackageIcon,
  RefreshCvIcon,
  SmartphoneIcon,
  TrendingDownIcon,
  VolumenIcon,
  AlignLeftIcon,
  BellOffIcon,
  CircleIcon,
  CreditCardIcon,
  FileMinusIcon,
  HelpCircleIcon,
  MapPinIcon,
  PaperclipIcon,
  RepeatIcon,
  SmileIcon,
  TrendingUpIcon,
  VolumenXIcon,
  AlignRightIcon,
  BluetoothIcon,
  ClipboardIcon,
  CropIcon,
  Maximize2Icon,
  PauseClrcleIcon,
  RewindIcon,
  SpeakerIcon,
  TriangleIcon,
  WatchIcon,
  AnchorIcon,
  BoldIcon,
  ClockIcon,
  CrosshairIcon,
  FilePlusIcon,
  HexagonIcon,
  MaximizeIcon,
  PauseIcon,
  RotateCcwIcon,
  SquareIcon,
  TruckIcon,
  WiffiOffIcon,
  ApertureIcon,
  BookIcon,
  CloudDrizzleIcon,
  CloudRainIcon,
  DatabaseIcon,
  FilmIcon,
  FileTextIcon,
  HomeIcon,
  MehIcon,
  PenTool,
  RotateCwIcon,
  StarIcon,
  SparksIcon,
  TvIcon,
  WifiIcon,
  ArchiveIcon,
  BookmarkIcon,
  CloudIcon,
  DeleteIcon,
  FilterIcon,
  IconInfo,
  MenuIcon,
  PercentIcon,
  RssIcon,
  StopCircleIcon,
  TwitchIcon,
  WindIcon,
  ArrowDownCircleIcon,
  BookOpenIcon,
  CloudLightingIcon,
  DiscIcon,
  FlagIcon,
  ImageIcon,
  MessageCircleIcon,
  PhoneCallIcon,
  SaveIcon,
  SunIcon,
  TwitterIcon,
  XCircleFillIcon,
  ArrowDownIcon,
  BoxIcon,
  CloudOffIcon,
  DivideCircleIcon,
  FolderIcon,
  InboxIcon,
  MessageSquareIcon,
  PhoneForwardedIcon,
  ScissorsIcon,
  SunriseIcon,
  TypeIcon,
  XCircleIcon,
  ArrowDownLeftIcon,
  BriefCaseIcon,
  CloudSnowIcon,
  DivideIcon,
  FolderMinusIcon,
  MicIcon,
  PhoneIcon,
  SearchIcon,
  SunsetIcon,
  UmrbrellaIcon,
  XIcon,
  ArrowDownRightIcon,
  CalendarIcon,
  CodeIcon,
  DivideSquareIcon,
  FolderPlusIcon,
  InstagramIcon,
  MicOffIcon,
  PhoneIncomingIcon,
  SendIcon,
  TableIcon,
  UnderlineIcon,
  XOctoganIcon,
  ArrowLeftCircleIcon,
  CameraIcon,
  CodepenIcon,
  DollarIcon,
  FramerIcon,
  ItalicICon,
  Minimize2Icon,
  MinimizeIcon,
  PhoneMissedIcon,
  SettingIcon,
  ServerIcon,
  TabletIcon,
  UnlockIcon,
  XSquareIcon,
  ArrowLeftIcon,
  CameraOffIcon,
  CodesandboxIcon,
  DownloadCloudIcon,
  DownloadIcon,
  FrownIcon,
  KeyIcon,
  MinusCircleIcon,
  PhoneOffIcon,
  Share2Icon,
  TagIcon,
  UploadCloudIcon,
  ChevronsLeftIcon,
  YoutubeIcon,
  ArrowRightCircleIcon,
  CastIcon,
  CoffeIcon,
  DribbbleIcon,
  GiftIcon,
  LayerIcon,
  MinusIcon,
  PhoneOutgoingIcon,
  ShareIcon,
  TargetIcon,
  UploadIcon,
  ZapIcon,
  ArrowRightIcon,
  CheckCircleFillIcon,
  ColumnsIcon,
  DropletIcon,
  GitBranchIcon,
  LayoutIcon,
  MinusSquareIcon,
  PieChartIcon,
  ShieldIcon,
  TerminalIcon,
  UserCheckIcon,
  ZapOffIcon,
  ArrowUpCircleIcon,
  CornerDownRightIcon,
  CheckCircleIcon,
  CommandIcon,
  Edit2Icon,
  GitCommitIcon,
  LifeBouyIcon,
  MonitorIcon,
  PlayCircleIcon,
  ShieldOffIcon,
  ThermometerIcon,
  UserMinusIcon,
  ZapOutIcon,
  ArrowUpIcon,
  CheckIcon,
  CompassIcon,
  Edit3Icon,
  GithubIcon,
  Link2Icon,
  MoonIcon,
  PlayIcon,
  ShoppingBagIcon,
  ThumbsDownIcon,
  UserPlusIcon,
  ZoomInIcon,
  ArrowUpLeftIcon,
  CheckSquareicon,
  CopyIcon,
  EditIcon,
  GitlabIcon,
  LinkedinIcon,
  MoreHorizontalIcon,
  PlusCircleIcon,
  ShoppingCartIcon,
  ThumbsUpIcon,
  UsersIcon,
  UserIcon,
};
