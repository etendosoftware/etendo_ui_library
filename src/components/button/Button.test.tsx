import React from 'react';
import renderer from 'react-test-renderer';
import { CheckIcon } from '../../assets/images/icons/CheckIcon';
import Button from './Button';

describe('Running Test for Input', () => {
  it('Check Button disabled', () => {
    const tree = renderer
      .create(
        <Button
          text={'Button CTA'}
          onPress={() => {}}
          typeStyle={'primary'}
          iconLeft={
            <CheckIcon
              fill={'white'}
              style={{
                marginRight: 10,
                width: 12,
                height: 10,
              }}
            />
          }
          disabled={true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
