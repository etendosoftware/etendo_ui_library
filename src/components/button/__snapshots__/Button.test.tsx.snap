// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running Test for Input Check Button disabled 1`] = `
<div
  className="css-view-175oi2r r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-cursor-1loqt21 r-touchAction-1otgn73"
  dir={null}
  onClick={[Function]}
  onContextMenu={[Function]}
  onKeyDown={[Function]}
  style={
    {
      "transitionDuration": "0s",
    }
  }
  tabIndex="0"
>
  <div
    className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
    dir={null}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(221,223,246,1.00)",
        "borderBottomLeftRadius": "5px",
        "borderBottomRightRadius": "5px",
        "borderTopLeftRadius": "5px",
        "borderTopRightRadius": "5px",
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "paddingBottom": "24px",
        "paddingLeft": "24px",
        "paddingRight": "24px",
        "paddingTop": "24px",
        "width": "auto",
      }
    }
    tabIndex={0}
  >
    <div
      className="css-view-175oi2r"
      dir={null}
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="10"
        bbWidth="12"
        fill="none"
        focusable={false}
        height={10}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "height": 10,
              "marginRight": 10,
              "width": 12,
            },
            {
              "flex": 0,
              "height": 10,
              "width": 12,
            },
          ]
        }
        vbHeight={32}
        vbWidth={43}
        width={12}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M41.0309 1.12469C39.3624 -0.37562 36.6578 -0.374175 34.9899 1.12469L15.4917 21.2185L6.88504 12.2221C5.21785 10.4783 2.51325 10.4783 0.92578 12.2224C-0.277994 13.9669 -0.277994 16.7949 0.92578 18.5386L12.4693 30.6925C14.1382 32.4358 16.8428 32.4358 18.5114 30.6925L41.0309 7.15158C42.6985 5.40676 42.6985 2.57913 41.0309 1.12469Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </div>
    <div
      className="css-text-146c3p1"
      dir="auto"
      style={
        {
          "color": "rgba(255,255,255,1.00)",
          "fontWeight": "bold",
        }
      }
    >
      Button CTA
    </div>
  </div>
</div>
`;
